### PR TITLE
fix(menu): align window and display coordinate spaces

### PR DESCRIPTION
**Pull request by steipete**
Fixes a coordinate-space mismatch in startup visibility recovery. The probe compared Cocoa screen frames (primary-display bottom-left origin) directly with Quartz window bounds (primary-display top-left origin). A lower monitor could hide a genuine blocked proxy, while a healthy window on an upper monitor could look blocked.

The injected probe now explicitly accepts primary-first screen frames and normalizes them before containment. The exact primary-origin blocked-proxy signature remains detectable when it overlaps the bottom of an upper display, while ordinary interior and top-edge windows keep their containment exemption. The same input boundary uses CoreGraphics’ rectangle decoder after checking numeric fields, removing the bespoke decoder. Existing displacement, on-screen, opt-in and startup-time guards are preserved. The change removes **11 production lines** and adds a 0.59.1 Unreleased note.

Validation:
- Eight baseline assertions fail across above/below/offset displays. The new tests exercise both window classification and the actual startup-recovery predicate.
- All 38 focused visibility tests pass, including the existing guard matrix and malformed rectangles. A further upper-display regression reproduced the review finding with two failed assertions, then passed with the narrow proxy exception; interior and top-edge windows remain protected. A malformed string fixture caught an Objective-C decoder exception during development; numeric validation now rejects it before decoding.
- `make check` passed with zero violations; independent P0–P2 review is clean.
- The final follow-up full suite passed all 1,069 selections / 90 groups first time, with no retries or timeouts (876.0 seconds), in a separate temporary preference home. An earlier revision had one recovered Claude PTY fixture retry.
- The superseded CI run was canceled. [Final-head CI](https://github.com/steipete/CodexBar/actions/runs/34639780555) passed before merge.

The coordinate contract follows [Cocoa screen coordinates](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CocoaDrawingGuide/Transforms/Transforms.html) and [CGWindow bounds](https://developer.apple.com/documentation/coregraphics/kcgwindowbounds). Proof is synthetic geometry through production decision code; this does not claim a live Tahoe reproduction or close the broader missing-pixel/position reports #3377 and #3355.

### CODE REVIEW

File: Sources/CodexBar/MenuBarStatusItemWindowProbe.swift [hunk 1]
```diff
@@ -39,16 +39,21 @@ enum MenuBarStatusItemWindowProbe {
...
         windowInfo: [[String: Any]],
-        displayBounds: [CGRect])
+        screenFrames: [CGRect])
         -> [MenuBarStatusItemWindowSnapshot]
     {
         guard !names.isEmpty else { return [] }
+        // Cocoa screen frames and Quartz window bounds have opposite vertical origins.
+        let primaryHeight = screenFrames.first?.height ?? 0
+        let displayBounds = screenFrames.map {
+            CGRect(x: $0.minX, y: primaryHeight - $0.maxY, width: $0.width, height: $0.height)
```
[the remark this answers is not in the document]
Comment (steipete) [reply Sources/CodexBar/MenuBarStatusItemWindowProbe.swift:57]: Fixed in 0b0c189df01182402e67c98ab3b36f2a3d60a11b. The new upper-display case reproduced failures in both proxy classification and the startup-recovery predicate. The fix retains the ordinary containment exemption and only recognizes the existing exact primary-origin proxy signature when it is not at the containing display’s top edge. Added upper-interior and top-edge cases guard against broader recreation. All 38 focused visibility tests, lint and independent P0–P2 review pass; full-suite and final-head CI gates are running.

**Comment by steipete** [comment]
Landed as 2a525d10003c82a8f05b71f83de73b690f5dce8c. The probe now normalizes Cocoa display frames before comparing Quartz window bounds, retaining the exact primary-origin blocked-proxy exception and the existing containment, displacement and startup guards. This fixes false or missed recovery with stacked displays while removing eleven production lines.

The review found an upper-display overlap edge case; two regression assertions reproduced it before the final correction. All 38 focused visibility tests, the final full suite (1,069 selections / 90 groups, zero retries or timeouts), format/lint and independent review passed. [Final-head CI](https://github.com/steipete/CodexBar/actions/runs/34639780555) passed across both macOS shards and Linux builds. An earlier local revision recovered one unrelated Claude PTY timeout; superseded CI was intentionally canceled for the review correction.

The main checkout is updated and clean, and its production/test files match the reviewed head. The Unreleased changelog is updated. This is geometry proof through the production recovery predicate, not a live reproduction of the broader missing-pixel or persisted-position reports; #3377 and #3355 remain open.
